### ISSUE TITLE
Bug fix to correctly pass “rate” metrics from Datadog Agent to Datadog is not working

### ISSUE THREAD

**Issue by m5i-ng** [vectordotdev/vector#14444]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

Metric submit via dogstatsd using datadog agent, metric type is stored as `rate` at Datadog without vector aggregator. 

With vector configuration, the metric type is stored as `count` at Datadog.

Bug fix mentioned in version 0.24.0 is not working

> Vector now has a concept of “rate” metrics which are a counter with an additional interval associated. This is used by the datadog_agent source the datadog_metrics sink to correctly pass “rate” metrics from the Datadog Agent to Datadog.

### Configuration

```text
data_dir: /vector-data-dir
    api:
      enabled: true
      address: 127.0.0.1:8686
      playground: false
    sources:
      datadog_agent:
        address: 0.0.0.0:8282
        type: datadog_agent
      fluent:
        address: 0.0.0.0:24224
        type: fluent
      internal_metrics:
        type: internal_metrics
      logstash:
        address: 0.0.0.0:5044
        type: logstash
      splunk_hec:
        address: 0.0.0.0:8080
        type: splunk_hec
      statsd:
        address: 0.0.0.0:8125
        mode: udp
        type: statsd
      syslog:
        address: 0.0.0.0:9000
        mode: tcp
        type: syslog
      vector:
        address: 0.0.0.0:6000
        type: vector
        version: "2"
    transforms:
      modify_datadog:
        type: remap
        inputs: [datadog_agent,statsd]
        source: |-
            {{- range .Values.remap_rules }}
            {{ . }}
            {{- end }}


    sinks:
      prom_exporter:
        type: prometheus_exporter
        inputs: [internal_metrics]
        address: 0.0.0.0:9090
      datadog_sink:
        type: datadog_metrics
        inputs:
          - modify_datadog
        {{- if .Values.vault.enabled }}
        api_key: $DD_API_KEY
        {{- else }}
        api_key: {{ .Values.apiKey }}
        {{- end }}
        healthcheck: true
        default_namespace: k8svector
```

### Version

0.24.0-debian

### Additional Context

Vector is running in Kubernetes cluster.

**Comment by mecsantos** [comment]
Hi @m5i-ng,

Regarding:
> Metric submit via dogstatsd using datadog agent, metric type is stored as rate at Datadog without vector aggregator.

Is the original metric submitted to dogstatsd a rate or a count? Looking at [Datadog docs](https://docs.datadoghq.com/metrics/types/?tab=count#submission-types-and-datadog-in-app-types), when you submit a count metric, it will become a rate https://a.cl.ly/d5unRgp8 so I'm suspecting that the original metric is a count. Now, if you send a count to vector, then the fixed mentioned in version 0.24.0 doesn't apply.

**Comment by jszwedko** [comment]
Fixed by https://github.com/vectordotdev/vector/pull/14741 . This'll be included in 0.24.2.
